build/build-in-docker

```diff
@@ -24,30 +24,39 @@ set -e
 SCRIPTDIR=$(cd $(dirname $0); pwd)
 
 LOCAL_MAVEN_REPO=${LOCAL_MAVEN_REPO:-"$HOME/.m2/repository"}
-CUDF_USE_PER_THREAD_DEFAULT_STREAM=${CUDF_USE_PER_THREAD_DEFAULT_STREAM:-ON}
 USE_GDS=${USE_GDS:-ON}
 export CMAKE_GENERATOR=${CMAKE_GENERATOR:-"Ninja"}
+export DOCKER_BUILD_EXTRA_ARGS="--platform=linux/amd64 --build-arg CMAKE_ARCH=x86_64"
+CUDA_VER=${CUDA_VER:-cuda11}
+USE_SANITIZER=${USE_SANITIZER:-ON}
+BUILD_FAULTINJ=${BUILD_FAULTINJ:-ON}
 
 if (( $# == 0 )); then
   echo "Usage: $0 <Maven build arguments>"
   exit 1
 fi
 
-_CUDF_CLEAN_SKIP=""
-# if ccache is enabled and libcudf.clean.skip not provided
-# by the user remove the cpp build directory
-#
-if [[ "$CCACHE_DISABLE" != "1" ]]; then
-  if [[ ! "$*" =~ " -Dlibcudf.clean.skip=" ]]; then
-    # Don't skip clean if ccache is enabled
-    # unless the user overrides
-    _CUDF_CLEAN_SKIP="-Dlibcudf.clean.skip=false"
-  fi
+case $(uname -m) in
+  x86_64|amd64)
+    arch=amd64;;
+  aarch64|arm64)
+    arch=arm64;;
+  *)
+    echo "Unsupported CPU architecture"; exit 1;;
+esac
+
+# Set env for arm64 build
+if [ "$arch" == "arm64" ]; then
+  profiles="${profiles},arm64"
```

Comment:
`profiles` appears to be unused.

Reply:
Actually, a better way to handle that profile is to update pom.xml so it automatically triggers based on the CPU architecture, that way users and scripts never need to specify it to get the right one, e.g:
```
  <profile>
    <id>arm64</id>
    <activation>
      <os>
        <arch>arm64</arch>
      </os>
    </activation>
    ....
```